### ISSUE TITLE
Question about the future of SafetyNet Bypassing: Play Integrity API

### ISSUE THREAD

**Issue by DavidBerdik** [kdrag0n/safetynet-fix#126]
Yesterday, I received the November 2021 Google Play newsletter in my inbox. While scrolling through it, this caught my eye:
![image](https://user-images.githubusercontent.com/7663225/143317756-73610046-bee4-422c-b624-f90cf0802ad1.png)

Needless to say, this intrigued me given my interest in bypassing SafetyNet, so I clicked on the button. Doing so redirected me [here](https://developer.android.com/google/play/integrity). This was my first time hearing about this, and unsurprisingly, it has made me a bit concerned about the future of SafetyNet bypassing. I realize that this post might be pointless, especially given that this API is still in an experimental stage, but I thought I would try asking here if anyone has any ideas about how this will impact the future of bypassing SafetyNet and in general, how this will impact the user experience on modified devices.

**Comment by Didgeridoohan** [comment]
Bypassing SafetyNet (or its replacement) has always been a temporary thing (although with an undetermined end date). As soon as Google starts to enforce the use of hardware backed key attestation (currently they allow devices to pass with basic attestation, which is why mods like this works) it is extremely unlikely that there will be any kind of bypass.

**Comment by DavidBerdik** [comment]
I was afraid that this was going to be the answer I was going to receive. ☹️

Here's hoping that this turns out to not be true. 🤞

**Comment by Didgeridoohan** [comment]
I'd say hoping for it not to be true is a hopeless activity... Take a look at [this tweet](https://twitter.com/shawnwillden/status/1463518016214163456?s=20) as an example (and note the occupation of it's author).

**Comment by DavidBerdik** [comment]
Yeah I pretty much figured that it is a hopeless activity, but let me dream! 😂 As for the Tweet you referenced, I have actually seen it before. Thanks for sharing though. 🙂 If anything, I posted here because I wanted to see if people more knowledgeable than me (who don't work for Google) had something worth sharing.

**Comment by undermark5** [comment]
I'd like to chime in my two cents here. Which may be completely worthless... Anyway, while hoping to that we will be able to bypass the Play Integrity API checks may be a lost cause, I don't think we have yet to give up on hoping it will be a better experience as Google may perhaps provide better education as to what the purpose of the Play Integrity API is for rather than developers adding it in as an easy "one size fits all fix" that safetynet has seemingly become. Personally, the application that I develop for my job was performing root checks using Rootbeer, but we have since decided to remove for a couple of reasons, one, it was super easy to bypass even without hiding root (assuming you know what rooting does and allows you to do and a basic understanding of basic application persistence), and two, the risk involved in allowing rooted users to use the application is primarily in an increased number of crashes, there is little if not nothing that they could do that wouldn't eventually result in some flags on the backend cropping up for that user. I pointed out that in the documentation and implementation guide for both safetynet and Play Integrity (we were considering implementing safetynet checks) were not to be used simply as root detection mechanisms but rather as a part of a larger analysis, one which we currently were not planning on doing.

So, while my two cents may not be worth much, I hope that with the eventual deprecation of Safetynet for Play Integrity developers and product managers are properly educated on the purpose that the API serves, providing a data point about the integrity of the system the application is running on, and ultimately recognize that simply disallowing a user access to the application/features because of a single data point is improper.

**Comment by DavidBerdik** [comment]
> So, while my two cents may not be worth much, I hope that with the eventual deprecation of Safetynet for Play Integrity developers and product managers are properly educated on the purpose that the API serves, providing a data point about the integrity of the system the application is running on, and ultimately recognize that simply disallowing a user access to the application/features because of a single data point is improper.

While I hope you are right, I honestly am not sure if I understand how it will help the rooting community if you do turn out to be right.

**Comment by undermark5** [comment]
It isn't something that we will benefit from as in we will be able to bypass the check or anything like that, it is instead developers and companies will actually implement proper security rather than just shoving a one size fits all "solution" where it doesn't belong, resulting in fewer applications preventing you from using them simply because you've rooted your device/unlocked your bootloader. This might just be my hopeful thinking, but Google has indicated in the overview of both Safetynet and Play Integrity that these checks are not to be used in a standalone manner, so hopefully education takes a higher priority this time.

**Comment by StakFallT** [comment]
Personally, I find the whole no-root allowed thing to be disgusting and distasteful. Who is -anyone- to tell anyone how to run their "computer". I mean think about it, it's like telling someone that bought a PC they are NOT allowed to have administrator access to their own machine and if they somehow have a means to achieve administrator access then sites will stop accepting payment, etc. It's so dumb and makes little sense. _Oh, so Amazon.com and other e-commerce sites won't let me be an admin on my own workstation? Oh kay... I guess I'll just suck it up and allow them power to dictate how I use my own device._  Stupid. Just plain stupid.
